.github/workflows/release.yml

```diff
@@ -5,82 +5,177 @@ permissions:
 
 on:
   workflow_dispatch:
+    inputs:
+      force:
+        required: false
+        description: 'If non-empty, extra force will be used push this release'
+        default: ''
+      commit:
+        required: false 
+        description: 'Fluvio git commit override (latest `master` by default)'
+        default: ''
+env:
+  USE_COMMIT: ${{ github.event.inputs.commit }}
+  FORCE_RELEASE: ${{ github.event.inputs.force }}
 
 jobs:
+  setup_job:
+    name: Setup workflow
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.version_step.outputs.VERSION }}
+      target_sha: ${{ steps.version_step.outputs.GIT_SHA }}
+      latest_tag: ${{ steps.docker_step.outputs.LATEST_TAG }}
+      release_tag: ${{ steps.docker_step.outputs.RELEASE_TAG }}
+    steps:
+      - name: Set target sha and Fluvio version
+        id: version_step
+        run: |
+          if [[ -z "${USE_COMMIT}" ]]; then
+            GITHUB_VERSION=$(curl -fsS https://raw.githubusercontent.com/infinyon/fluvio/${{ github.sha }}/VERSION)
```

Comment:
I don't think this will work unless you use `export GITHUB_VERION=...`

Reply:
I don't need the var to leave the job, this works for reusing in the if/else context. I'm using the `set-output` to export to other jobs.. This does work: https://github.com/infinyon/fluvio-gh-actions-dev/runs/2507309943?check_suite_focus=true

Reply:
Right, but even just in regular bash scripting, a variable does not typically persist longer than a single command, right? Like if I did `RUST_LOG=debug fluvio version; fluvio version` only the first one would have the env variable. I'd need to do `export RUST_LOG=debug; fluvio version; fluvio version` in order for it to span multiple commands

Reply:
That's correct. I guess the reason it doesn't matter in this case is that I'm not spawning a child process where I need these vars to be visible. You can try this in your shell too.

```
$ UNEXPORTED="true"
$ echo "Can I see the var? ${UNEXPORTED}"
Can I see the var? true
```

Reply:
I just went ahead and added the export bc it doesn't hurt, nor does it change the behavior.